persistent-postgresql/test/CustomConstraintTest.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE EmptyDataDecls             #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE QuasiQuotes                #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UndecidableInstances       #-}
+module CustomConstraintTest where
+
+import PgInit
+import Data.List (find)
+import qualified Data.Text as T
+
+share [mkPersist sqlSettings, mkMigrate "customConstraintMigrate"] [persistLowerCase|
+CustomConstraint1
+    some_field Text
+    deriving Show
+
+CustomConstraint2
+    cc_id CustomConstraint1Id constraint=custom_constraint
+    deriving Show
+|]
+
+alterQuery :: Text
+alterQuery = "ALTER TABLE \"custom_constraint2\" ADD CONSTRAINT \"custom_constraint\" FOREIGN KEY(\"cc_id\") REFERENCES \"custom_constraint1\"(\"id\")"
+
+specs :: (MonadIO m, MonadFail m) => RunDb SqlBackend m -> Spec
+specs runDb = do
+  describe "custom constraint used in migration" $ do
+    it "alter table uses custom constraint name" $ runDb $ do
+      ms <- getMigration customConstraintMigrate
+      let alter = find (== alterQuery) ms
```

Comment:
This seems a bit hacky, but I couldn't think of a better way to ensure that the migration uses the correct constraint name.

@parsonsmatt Thoughts?

Reply:
I ended up removing this test case as it makes the tests non-idempotent. Now we just ensure the constraint exists after running the migration.